app/services/audit/wasapi_warc_lister.rb

```diff
@@ -11,17 +11,18 @@ def initialize(wasapi_collection_id:, wasapi_provider:, wasapi_account:, embargo
       @embargo_months = embargo_months
     end
 
+    # Iterate through each WARC filename, or return as a list
     def each(&block)
-      if block_given?
-        next_url = first_page_url
-        while next_url
-          response = get_page(next_url)
-          files_for(response).each { |file| block.call(file) }
-          next_url = response[:next]
-        end
-      else
-        to_enum(:each)
-      end
+      return to_enum(:each) unless block_given?
+
+      files(yield_object: false, &block)
+    end
+
+    # Iterate through WARC file metadata, or return as a list
+    def each_object(&block)
+      return to_enum(:each_object) unless block_given?
+
+      files(&block)
     end
```

Comment:
I wasn't sure if the above `each_object` was the best name for the method. I basically wanted to reuse the machinery for fetching the response from the Archive-It API, but optionally return a hash of metadata instead of only returning the filename.

Reply:
`each_warc_metadata`?

Reply:
Gotcha. I wonder if it's worth considering wiring up `#each` so that it returns the object all the time, and using `.map { |file| file[:filename] }` wherever we only want the filenames? That'd remove some machinery, at least.

Reply:
Oh yeah, that's a good point. I'll see where that's happening.